src/main/java/io/jenkins/plugins/launchable/Ingester.java

```diff
@@ -52,22 +50,25 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
 
             if (apiKey==null)     return; // not yet configured
 
+            OrganizationWorkspace orgWs = OrganizationWorkspace.fromApiKey(apiKey.getPlainText());
+
             // attempted to use JDK HttpRequest, but gave up due to the lack of multipart support
             // TODO: how do I obtain a properly configured HttpClient for the proxy setting in Jenkins?
             try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                String endpoint = System.getenv("INSIGHT_UPLOAD_URL") ;
 
-                String endpoint = System.getenv("INSIGHT_UPLOAD_URL");
                 if (endpoint==null) {
                     endpoint = DEFAULT_UPLOAD_URL;
                 }
-                var hc = new HttpPost(endpoint);
+                HttpPost hc = new HttpPost(String.format("%s/intake/organizations/%s/workspaces/%s/events/jenkins", endpoint, orgWs.getOrganization(), orgWs.getWorkspace()));
 
-                var builder = MultipartEntityBuilder.create();
+                MultipartEntityBuilder builder = MultipartEntityBuilder.create();
                 builder.addTextBody("metadata", properties.build().toString(), ContentType.APPLICATION_JSON);
                 builder.addBinaryBody("file", report, ContentType.APPLICATION_XML, "junitResult.xml");
 
                 hc.setEntity(new GzipCompressingEntity(builder.build()));
                 hc.addHeader("Authorization", "Bearer " + apiKey.getPlainText());
+                hc.setHeader(HttpHeaders.CONTENT_ENCODING, "gzip");
```

Comment:
We talked it this part directly, it seems server side can't handle compressed request correctly.